File: IdeaProjects/Tienda/src/Menu/Cliente.java
package Menu;

import Productos.Producto;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.Scanner;


public class Cliente {
    Scanner leer = new Scanner(System.in);
    ArrayList<Producto> comprado = new ArrayList<>();
    ArrayList<Integer> cantidad = new ArrayList<>();
    ArrayList<LocalDate> fecha = new ArrayList<>();
    String nombre;
    LocalDate fechaActual;
    int edad, id;

    public Cliente(String nombre, int edad) {
        this.edad = edad;
        this.nombre = nombre;
        id = 0;
    }

    public int getEdad() {
        return edad;
    }

    public void setEdad(int edad) {
        this.edad = edad;
    }

    public String getNombre() {
        return nombre;
    }

    public void setNombre(String nombre) {
        this.nombre = nombre;
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public void registrarProducto(Producto producto, int cantidad) {
        this.fechaActual = LocalDate.now();
        fecha.add(fechaActual);
        comprado.add(producto);
        this.cantidad.add(cantidad);
    }

    public void mostrarProductos() {
        int i = 0;
        LocalDate fechaAnterior = LocalDate.of(1996, 1, 1);
        for (Producto producto : comprado) {
            if (!fechaAnterior.isEqual(fecha.get(i))) {
                System.out.println("\nFecha de compra(s): " + fecha.get(i) + "\n");
            }
            fechaAnterior = fecha.get(i);
            System.out.println("Nombre: " + producto.getName());
            System.out.println("id: " + producto.getId());
            System.out.println("Número de serie: " + producto.getNo());
            System.out.println("Fecha de importación: " + producto.getFechaImportacion());
            System.out.println("Precio: " + producto.getPrecio());
            System.out.println("Clase: " + producto.getClass());
            System.out.println("Cantidad comprada: " + this.cantidad.get(i));
            System.out.println("Precio total: " + (producto.getPrecio() * cantidad.get(i)));
            i++;
        }
    }



    @Override
    public String toString() {
        return " Nombre='" + nombre + '\'' +
                ", Edad=" + edad +
                ", Id=" + id +
                '}';
    }
}

File: IdeaProjects/Tienda_py/.idea/Menu/Cliente.py
from datetime import date
from datetime import datetime

class Cliente:
    def _init_(self, nombre, edad):
        self.nombre = nombre
        self.edad = edad
        self.id = 0
        self.comprado = []
        self.cantidad = []
        self.fecha = []

    def registrar_producto(self, producto, cantidad):
        self.fecha_actual = date.today()
        self.fecha.append(self.fecha_actual)
        self.comprado.append(producto)
        self.cantidad.append(cantidad)

    def mostrar_productos(self):
        i = 0
        fecha_anterior = date(1996, 1, 1)
        for producto in self.comprado:
            if self.fecha[i] != fecha_anterior:
                print("\nFecha de compra(s):", self.fecha[i], "\n")
            fecha_anterior = self.fecha[i]
            print("Nombre:", producto.nombre)
            print("id:", producto.id)
            print("Número de serie:", producto.numero_serie)
            print("Fecha de importación:", producto.fecha_importacion)
            print("Precio:", producto.precio)
            print("Clase:", producto._class.name_)
            print("Cantidad comprada:", self.cantidad[i])
            print("Precio total:", producto.precio * self.cantidad[i])
            i += 1

    def _str_(self):
        return f"Nombre='{self.nombre}', Edad={self.edad}, Id={self.id}"


